week3/class_version/cv_tictactoe_board.py

```diff
@@ -0,0 +1,207 @@
+"""
+Module sets up the TicTacToeBoard class that can be used
+to play tictactoe
+"""
+
+EMPTY = 0
+PLAYERX = 1
+PLAYERO = 2
+DRAW = -2
+
+def switch_player(player):
+    if player == PLAYERO:
+        return PLAYERX
+    return PLAYERO
+
+class TicTacToeBoard(object):
+    """
+    Class to represent a Tic-Tac-Toe board.
+    """
+
+    def __init__(self, dimension, reverse=False, board=None):
+        """
+        Initialize the TTTBoard object with the given dimension and
+        whether or not the game should be reversed.
+        """
+        self._dimension = dimension
+        self._reverse = reverse
+        self._board = self.new_board()
+
+    def __str__(self):
+        """
+        Human readable representation of the board.
+        """
+        dashes = '---'*self.dimension + '-'*(self.dimension-1)
+
+        replacements = [('[[', ' '), ('[', ''),
+                        ('],', '\n'+dashes+'\n'), (', ', ' | '),
+                        (']]', ''), ('0', ' '),
+                        ('1', 'X'), ('2', 'O')]
+
+        board_string = str(self.board)
+        for index in replacements:
+            board_string = board_string.replace(index[0], index[1])
+
+        return board_string
+
+    @property
+    def dimension(self):
+        """
+        Return the dimension of the board.
+        """
+        return self._dimension
+
+    @dimension.setter
+    def dimension(self, value):
```

Comment:
I didn't think this was possible in Python: two functions with the same name but different signatures.

Does this work as intended?